From e425d46f735e6ccb8c83237f47b4bf338d317a84 Mon Sep 17 00:00:00 2001
From: Son Luong Ngoc <sluongng@gmail.com>
Date: Tue, 29 Apr 2025 23:41:18 +0200
Subject: [PATCH] go_grpc_compiler: default to go_grpc_v2 (#2071)

Set the default grpc compiler to go_grpc_v2 because go_grpc is
considered deprecated.


https://github.com/bazel-contrib/rules_go/blob/8251e45ee86fd17f3fedb34caa2d06a40089f22e/proto/BUILD.bazel#L34
---
 cmd/gazelle/integration_test.go                             | 4 ++--
 language/go/config.go                                       | 2 +-
 language/go/constants.go                                    | 2 +-
 language/go/fix_test.go                                     | 2 +-
 language/go/testdata/proto_file_mode/abc/BUILD.want         | 2 +-
 language/go/testdata/service/BUILD.want                     | 2 +-
 .../go/testdata/service_gogo_subdir_reset/sub/BUILD.want    | 2 +-
 8 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/cmd/gazelle/integration_test.go b/cmd/gazelle/integration_test.go
index e82d6af30..b2505c021 100644
--- a/cmd/gazelle/integration_test.go
+++ b/cmd/gazelle/integration_test.go
@@ -771,7 +771,7 @@ proto_library(
 
 go_proto_library(
     name = "repo_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "example.com/repo",
     proto = ":repo_proto",
     visibility = ["//visibility:public"],
@@ -2033,7 +2033,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
 go_proto_library(
     name = "service_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "example.com/repo/service",
     proto = ":service_proto",
     visibility = ["//visibility:public"],
diff --git a/language/go/config.go b/language/go/config.go
index ac7f089ff..2852180aa 100644
--- a/language/go/config.go
+++ b/language/go/config.go
@@ -148,7 +148,7 @@ const (
 
 var (
 	defaultGoProtoCompilers = []string{"@io_bazel_rules_go//proto:go_proto"}
-	defaultGoGrpcCompilers  = []string{"@io_bazel_rules_go//proto:go_grpc"}
+	defaultGoGrpcCompilers  = []string{"@io_bazel_rules_go//proto:go_grpc_v2"}
 )
 
 func (m testMode) String() string {
diff --git a/language/go/constants.go b/language/go/constants.go
index 0affc48a9..dfb75b466 100644
--- a/language/go/constants.go
+++ b/language/go/constants.go
@@ -33,7 +33,7 @@ const (
 
 	// grpcCompilerLabel is the label for the gRPC compiler plugin, used in the
 	// "compilers" attribute of go_proto_library rules.
-	grpcCompilerLabel = "@io_bazel_rules_go//proto:go_grpc"
+	grpcCompilerLabel = "@io_bazel_rules_go//proto:go_grpc_v2"
 
 	// goProtoSuffix is the suffix applied to the labels of all generated
 	// go_proto_library targets.
diff --git a/language/go/fix_test.go b/language/go/fix_test.go
index d811102ac..fd5e8687a 100644
--- a/language/go/fix_test.go
+++ b/language/go/fix_test.go
@@ -581,7 +581,7 @@ proto_library(
 
 go_proto_library(
     name = "foo_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "example.com/repo",
     proto = ":foo_proto",
     visibility = ["//visibility:public"],
diff --git a/language/go/testdata/proto_file_mode/abc/BUILD.want b/language/go/testdata/proto_file_mode/abc/BUILD.want
index 8d6588bc2..24243d1e8 100644
--- a/language/go/testdata/proto_file_mode/abc/BUILD.want
+++ b/language/go/testdata/proto_file_mode/abc/BUILD.want
@@ -21,7 +21,7 @@ go_proto_library(
         "abc/a.proto",
         "xyz/x.proto",
     ],
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "example.com/repo/proto_file_mode/abc",
     protos = [
         ":a_proto",
diff --git a/language/go/testdata/service/BUILD.want b/language/go/testdata/service/BUILD.want
index abdda81e6..17f24b17f 100644
--- a/language/go/testdata/service/BUILD.want
+++ b/language/go/testdata/service/BUILD.want
@@ -18,7 +18,7 @@ go_proto_library(
         "google/protobuf/any.proto",
         "service/sub/sub.proto",
     ],
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "example.com/repo/service",
     proto = ":service_proto",
     visibility = ["//visibility:public"],
diff --git a/language/go/testdata/service_gogo_subdir_reset/sub/BUILD.want b/language/go/testdata/service_gogo_subdir_reset/sub/BUILD.want
index bdf1ccd81..7ece1724e 100644
--- a/language/go/testdata/service_gogo_subdir_reset/sub/BUILD.want
+++ b/language/go/testdata/service_gogo_subdir_reset/sub/BUILD.want
@@ -12,7 +12,7 @@ proto_library(
 go_proto_library(
     name = "protos_gogo_go_proto",
     _gazelle_imports = [],
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = ["@io_bazel_rules_go//proto:go_grpc_v2"],
     importpath = "example.com/repo/protos_gogo",
     proto = ":protos_gogo_proto",
     visibility = ["//visibility:public"],
